docs/src/main/asciidoc/spring-data-jpa-guide.adoc

```diff
@@ -0,0 +1,592 @@
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven 3.5.3+
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `using-spring-data-jpa` directory.
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=using-spring-data \
+    -DclassName="org.acme.spring.data.jpa.FruitResource" \
+    -Dpath="/greeting" \
+    -Dextensions="spring-data-jpa,jaxrs-json,quarkus-jdbc-postgresql"
+----
+
+This command generates a Maven project with a REST endpoint and imports the `spring-data-jpa` extension.
+
+== Define the Entity
+
+Throughout the course of this guide, we will be using the following JPA Entity:
+
+[source,java]
+----
+package org.acme.spring.data.jpa;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+@Entity
+public class Fruit {
+
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    private String name;
+
+    private String color;
+
+
+    public Fruit() {
+    }
+
+    public Fruit(String name, String color) {
+        this.name = name;
+        this.color = color;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+}
+----
+
+
+== Configure database access properties
+
+Add the following properties to `application.properties` to configure access to a local PostgreSQL.
+
+[source,properties]
+----
+quarkus.datasource.url=jdbc:postgresql:quarkus_test
+quarkus.datasource.driver=org.postgresql.Driver
+quarkus.datasource.username=quarkus_test
+quarkus.datasource.password=quarkus_test
+quarkus.datasource.max-size=8
+quarkus.datasource.min-size=2
+quarkus.hibernate-orm.database.generation=drop-and-create
+----
+
+This configuration assumes that PostgreSQL will be running locally.
+
+A very easy way to accomplish that is by using the following docker command:
+
+```
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 --name quarkus_test -e POSTGRES_USER=quarkus_test -e POSTGRES_PASSWORD=quarkus_test -e POSTGRES_DB=quarkus_test -p 5432:5432 postgres:10.5
```

Comment:
Maybe use `11.5` instead?

Reply:
Again, I lifted this from the Panache guide, so perhaps we should stick with the same version for now and then update them all in a next step?

Reply:
So, my advice would be to use the latest available when writing a new guide because you're testing it with a given version. You can still revisit the old guide later on.

Otherwise, you end up using the same old version on all guides.

Moreover, I don't think we need consistency given the users will use one or the other. But maybe Docker won't like it if you use the same instance name for another version? Considering you have `-rm=true`, I would expect this image to be short lived though.

Reply:
Fixed